File: 502. IPO/Java/Solution.java
public class Solution {
    public int findMaximizedCapital(int k, int W, int[] Profits, int[] Capital) {
        PriorityQueue<int[]> pqCapical = new PriorityQueue<>(new Comparator<int[]>() {
            @Override
            public int compare(int[] o1, int[] o2) {
                return o1[0] - o2[0];
            }
        });
        PriorityQueue<Integer> pqProfit = new PriorityQueue<>();
        for (int i = 0; i < Profits.length; i++) {
            pqCapical.add(new int[] {Capital[i], Profits[i]});
        }
        for (int i = 0; i < k; i++) {
            while (!pqCapical.isEmpty() && pqCapical.peek()[0] <= W) {
                int[] item = pqCapical.poll();
                pqProfit.add(-item[1]);
            }
            if (pqProfit.isEmpty()) {
                break;
            }
            W += -pqProfit.poll();
        }
        return W;
    }
}

File: 502. IPO/C++/Solution.cpp
class Solution {
public:
    int findMaximizedCapital(int k, int W, vector<int> &Profits, vector<int> &Capital) {
        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> pqCapital;
        priority_queue<int> pqProfit;
        for (int i = 0; i < Profits.size(); i++) {
            pqCapital.push(pair<int, int>(Capital[i], Profits[i]));
        }
        for (int i = 0; i < k; i++) {
            while (!pqCapital.empty() && pqCapital.top().first <= W) {
                pqProfit.push(pqCapital.top().second);
                pqCapital.pop();
            }
            if (pqProfit.empty()) {
                break;
            }
            W += pqProfit.top();
            pqProfit.pop();
        }
        return W;
    }
};